main.cc

```diff
@@ -1151,6 +1153,28 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
             scheduling_group_key_config cql_sg_stats_cfg = make_scheduling_group_key_config<cql_transport::cql_sg_stats>(maintenance_socket_enabled::no);
             auto cql_sg_stats_key = scheduling_group_key_create(cql_sg_stats_cfg).get();
 
+            supervisor::notify("starting disk space monitor");
+            auto dsm_cfg = utils::disk_space_monitor<lowres_clock>::config{
+                .sched_group = dbcfg.compaction_scheduling_group,
+                .normal_polling_interval = cfg->disk_space_monitor_normal_polling_interval_in_seconds,
+                .high_polling_interval = cfg->disk_space_monitor_high_polling_interval_in_seconds,
+                .polling_interval_threshold = cfg->disk_space_monitor_polling_interval_threshold,
+            };
+            fs::path data_dir;
+            if (data_dir_set.get_paths().size() == 0) {
```

Comment:
nit, i'd suggest just put:
```c++
if (data_dir_set.empty()) {
    throw ...;
}
fs::path data_dir = *data_dir_set.get_paths().begin();
```

because

- if we don't care about the exact number of elements in the return value `get_paths()`, we can just check for its emptiness
- less indent level, as the branch with `throw` is practically an early return.
- smaller lexical scope of `data_dir` for better readability.

Reply:
done in v8